src/attribute_type.rs

```diff
@@ -412,22 +418,18 @@ impl SdpAttributeRtcp {
         }
     }
 
-    fn set_addr(&mut self, addr: IpAddr) {
+    fn set_addr(&mut self, addr: ExplicitlyTypedAddress) {
         self.unicast_addr = Some(addr)
     }
 }
 
-impl ToString for SdpAttributeRtcp {
-    fn to_string(&self) -> String {
-        let unicast_addr_str_opt = match self.unicast_addr {
-            None => None,
-            Some(x) => Some(address_to_string(x)),
-        };
-        format!(
-            "{port}{unicast_addr}",
-            port = self.port.to_string(),
-            unicast_addr = option_to_string!(" {}", unicast_addr_str_opt)
-        )
+impl fmt::Display for SdpAttributeRtcp {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Some(addr) = self.unicast_addr.clone() {
```

Comment:
What is the clone() needed for? Isn't a reference good enough for write!()?

Reply:
Fixed.